### ISSUE TITLE
ComputeOverlayIntersection not behaving correctly with dashboard overlay

### ISSUE THREAD

**Issue by borishsu** [ValveSoftware/openvr#1239]
#### Issue description
The behavior of API ComputeOverlayIntersection() with dashboard overlay changed when updating SteamVR from 1.7.15 to 1.8.20.

#### Steps to reproduce the issue

1. Create a dashboard overlay
2. Use ComputeOverlayIntersection() and the pose of any controller to get the intersection results
3. Print out the VROverlayIntersectionResults_t.vUVs
4. Run with SteamVR 1.7.15 and 1.8.20

#### Expected result

- ComputeOverlayIntersection() should work the same at any position.
- The vUVs returned should be the same when pointing at the same position.

#### Actual result

- When running with SteamVR 1.8.20, ComputeOverlayIntersection() doesn't return true when pointing at upper half the dashboard.
- When pointing out of the dashboard (below it), ComputeOverlayIntersection() still returns true. It seems that the intersection area is shifted down along y-axis.

#### Screenshots
![steamvr_1 7 15](https://user-images.githubusercontent.com/38482575/68655546-2a726100-056b-11ea-8040-1a0d4008dc29.png)
![steamvr_1 8 20](https://user-images.githubusercontent.com/38482575/68655553-2e05e800-056b-11ea-8245-4b7d8f28428d.png)
![2019-11-12 16_40_14](https://user-images.githubusercontent.com/38482575/68655555-2f371500-056b-11ea-9393-0eef7d5228db.png)

Regards,
Boris

**Comment by jeremyselan** [comment]
Thanks, we're looking into it.

**Comment by jeremyselan** [comment]
Are you able to share this test app?

**Comment by borishsu** [comment]
Hi Jeremy,

Thanks for your reply. Here is the test app: [TestDashboardOverlay.zip](https://github.com/ValveSoftware/openvr/files/3850737/TestDashboardOverlay.zip)

The data to fill in VROverlayIntersectionParams_t is hard-coded to the right controller. The render texture sent is 1920 * 1080 with the width 3 meters. Please let me know if any further information is required.

Regards,
Boris

**Comment by borishsu** [comment]
Sorry for accidentally clicking the close button. The issue is reopened.

**Comment by jeremyselan** [comment]
Thanks for the example.  When I test it though, it fails on all versions (including 1.7.15, 1.6.X, etc).  Which HMD are you using?  Are you able to share the source code for this app?

**Comment by jeremyselan** [comment]
Actually, I'm able to repo both cases now, don't worry about source code.

**Comment by jeremyselan** [comment]
Ok, can you give this a test in 1.9.3 (just published SteamVR[beta]), and confirm it works?  I believe we've resolved it.

Curiously, this still has a consistent 'forward orientation' bug for Knuckles controllers (though it now works for Vive / Vive Pro ).   This is unexpected. If you're willing to share source, we're happy to take a look.   Perhaps if you draw the ray for the controller representing the direction / point you're passing to ComputeOverlayIntersection, it would make the knuckles issue more obvious?

**Comment by borishsu** [comment]
Thank you for the help! I have tested in beta 1.9.3 and it works perfectly. Will this fix go in the hotfix of 1.8.20?

Here is the code I used in the test app. I didn't use the transform of the "tip" render model component as the ray to pass to `ComputeOverlayIntersection`. Instead I just picked the position and the forward direction of the returned pose. Would this be the cause of the inconsistent forward orientation?

```csharp
var rightIndex = OpenVR.System.GetTrackedDeviceIndexForControllerRole(ETrackedControllerRole.RightHand);
var err = OpenVR.Compositor.GetLastPoseForTrackedDeviceIndex(rightIndex, ref pose, ref gamePose);
if (err == EVRCompositorError.None)
{
	var rightRigidTransform = new SteamVR_Utils.RigidTransform(pose.mDeviceToAbsoluteTracking);
	rightGameObject.transform.position = rightRigidTransform.pos;
	rightGameObject.transform.rotation = rightRigidTransform.rot;

	var input = new VROverlayIntersectionParams_t();

	input.vSource.v0 = rightGameObject.transform.position.x;
	input.vSource.v1 = rightGameObject.transform.position.y;
	input.vSource.v2 = -rightGameObject.transform.position.z;
	input.vDirection.v0 = rightGameObject.transform.forward.x;
	input.vDirection.v1 = rightGameObject.transform.forward.y;
	input.vDirection.v2 = -rightGameObject.transform.forward.z;
	input.eOrigin = OpenVR.Compositor.GetTrackingSpace();
	
	var output = new VROverlayIntersectionResults_t();
	if (OpenVR.Overlay.ComputeOverlayIntersection(overlayHandle, ref input, ref output))
	{
		// ...
	}
}
```

Regards,
Boris
